src/form/input/Input.tsx

```diff
@@ -118,7 +130,11 @@ function Input(props: InputProps) {
       if (maxFractionDigits > 0) {
         const decimalNumberParts = newValue.split(DECIMAL_NUMBER_SEPARATOR);
         const decimalPart = decimalNumberParts[1];
-        const integerPart = decimalNumberParts[0].replace(THOUSANDTHS_SEPARATOR, "");
+        const integerPart = decimalNumberParts[0];
```

Comment:
As it was discussed during the weekly meeting, `onChange` handler should `parse` the input value into a real number. So I believe `formattedNewValue` should just be computed using `parseNumber` utility